challenges/2021-04-13-single-and-ready-to-mingle/solutions/doublebridges.js

```diff
@@ -0,0 +1,44 @@
+// # Single and Ready to Mingle
+// 
+// Given an array of integers in which two elements appear exactly once and all 
+// other elements appear exactly twice, find the two elements that appear only 
+// once.
+// 
+// ## Business Rules/Errata
+// 
+// - You can assume that the input will **always** include at least two and 
+//   exactly two elements that appear only once.
+// - You may not assume that the list will be sorted ahead of time.
+// - Extra Challenge: Can you complete this puzzle in linear time and constant 
+//   space?
+// - Extra Extra Challenge: Can you make your solution generic over other input 
+//   types?
+// - Extra Extra Extra Challenge: Do it in TypeScript.
+// 
+// ## Examples
+// 
+// ```
+// input = [2, 4, 6, 8, 10, 2, 6, 10];
+// findSingles(input)  // [4, 8]
+// ```
+// ```
+// input = [1, 2, 3, 2, 1, 4, 4, 6, 7, 8, 7, 6, 9, 9];
+// findSingles(input)  // [3, 8]
+// ```
+
+function findSingles(input) {
+  let ansArr = [];
+  for (let idx in input) {
+    let val = input[idx]
+    if (!ansArr.includes(val)) {
+      ansArr.push(val)
+    } else {
+      ansArr.splice(ansArr.indexOf(val), 1)
```

Comment:
I'm not sure what the efficiency of splice operations are in JS, but typically this would mean:
- Finding the index in `ansArr`; O(1), probably
- Shifting everything to the right of that index over by one; O(n)
- Setting the internal tracking value of `ansArr`'s length to the new length; O(1)

Reply:
You're right. I wasn't taking the method calls inside the loop in consideration. I was just thinking about the one pass loop and the one extra array created.